# First Steps with Python and Jupyter 

![](https://i.imgur.com/gvSnw4A.png)

This tutorial covers the following topics:

* Performing arithmetic operations using Python
* Solving multi-step problems using variables
* Evaluating conditions using Python
* Combining conditions with logical operators
* Adding text styles using Markdown

## Performing Arithmetic Operations using Python

Let's begin by using Python as a calculator. You can write and execute Python using a code cell within Jupyter. 

> **Working with cells**: To create a new cell within Jupyter, you can select "Insert > Insert Cell Below" from the menu bar or just press the "+" button on the toolbar. You can also use the keyboard shortcut `Esc+B` to create a new cell. Once a cell is created, click on it to select it. You can then change the cell type to code or markdown (text) using the "Cell > Cell Type" menu option. You can also use the keyboard shortcuts `Esc+Y` and `Esc+M`. Double-click a cell to edit the content within the cell. To apply your changes and run a cell, use the "Cell > Run Cells" menu option or click the "Run" button on the toolbar or just use the keyboard shortcut `Shift+Enter`. You can see a full list of keyboard shortcuts using the "Help > Keyboard Shortcuts" menu option.

Run the code cells below to perform calculations and view their result. Try changing the numbers and run the modified cells again to see updated results. Can you guess what the `//`, `%`, and `**` operators are used for?

In [5]:
2 + 3 + 9

14

In [6]:
99 - 73

26

In [7]:
23.54 * -1432

-33709.28

In [8]:
100 / 7

14.285714285714286

In [9]:
100 // 7

14

In [10]:
100 % 7

2

In [11]:
5 ** 3

125

In [12]:
((2 + 5) * (17 - 3)) / (4 ** 3)

1.53125

Python supports the following arithmetic operators:

| Operator   | Purpose           | Example     | Result    |
|------------|-------------------|-------------|-----------|
| `+`        | Addition          | `2 + 3`     | `5`       |
| `-`        | Subtraction       | `3 - 2`     | `1`       |
| `*`        | Multiplication    | `8 * 12`    | `96`      |
| `/`        | Division          | `100 / 7`   | `14.28..` |
| `//`       | Floor Division    | `100 // 7`  | `14`      |    
| `%`        | Modulus/Remainder | `100 % 7`   | `2`       |
| `**`       | Exponent          | `5 ** 3`    | `125`     |

## Solving multi-step problems using variables

Let's try solving the following word problem using Python: 

> A grocery store sells a bag of ice for $1.25 and makes a 20% profit. If it sells 500 bags of ice, how much total profit does it make?

We can list out the information provided and gradually convert the word problem into a mathematical expression that can be evaluated using Python. 

*Cost of ice bag ($)* = 1.25

*Profit margin* = 20% = .2

*Profit per bag ($)* = profit margin * cost of ice bag = .2 * 1.25

*No. of bags* = 500

*Total profit* = no. of bags * profit per bag = 500 * (.2 * 1.25)

In [13]:
500 * (.2 * 1.25)

125.0

In [14]:
cost_of_ice_bag = 1.25

In [15]:
profit_margin = .2

In [16]:
number_of_bags = 500

In [17]:
profit_margin

0.2

In [18]:
profit_per_bag = cost_of_ice_bag * profit_margin

In [19]:
profit_per_bag

0.25

In [20]:
total_profit = number_of_bags * profit_per_bag

In [21]:
total_profit

125.0

In [22]:
net_profit

NameError: name 'net_profit' is not defined

In [23]:
print("The grocery store makes a total profit of $", total_profit)

The grocery store makes a total profit of $ 125.0


> **`print`**: The `print` function is used to display information. It takes one or more inputs, which can be text (within quotes, e.g., `"this is some text"`), numbers, variables, mathematical expressions, etc. We'll learn more about variables & functions in the next tutorial.

Creating a code cell for each variable or mathematical operation can get tedious. Fortunately, Jupyter allows you to write multiple lines of code within a single code cell. The result of the last line of code within the cell is displayed as the output. 

Let's rewrite the solution to our word problem within a single cell.

In [24]:
# Store input data in variables
cost_of_ice_bag = 1.25
profit_margin = .2
number_of_bags = 500

# Perform the required calculations
profit_per_bag = cost_of_ice_bag * profit_margin
total_profit = number_of_bags * profit_per_bag

# Display the result
print("The grocery store makes a total profit of $", total_profit)

The grocery store makes a total profit of $ 125.0


Note that we're using the `#` character to add *comments* within our code. 

> **Comments**: Comments and blank lines are ignored during execution, but they are useful for providing information to humans (including yourself) about what the code does. Comments can be inline (at the end of some code), on a separate line, or even span multiple lines. 

Inline and single-line comments start with `#`, whereas multi-line comments begin and end with three quotes, i.e. `"""`. Here are some examples of code comments:

In [25]:
my_favorite_number = 1 # an inline comment

In [26]:
"""This is a multi-line comment.
Write as little or as much as you'd like.

Comments are really helpful for people reading
your code, but try to keep them short & to-the-point.

Also, if you use good variable names, then your code is
often self explanatory, and you may not even need comments!
"""
a_neutral_number = 5



> **EXERCISE**: A travel company wants to fly a plane to the Bahamas. Flying the plane costs 5000 dollars. So far, 29 people have signed up for the trip. If the company charges 200 dollars per ticket, what is the profit made by the company? Create variables for each numeric quantity and use appropriate arithmetic operations.

In [60]:
plane_cost=5000
total_people_signed_up=29
ticket_cost=200

In [61]:
amount_received_by_company=total_people_signed_up*ticket_cost

In [62]:
total_profit=amount_received_by_company-plane_cost

In [63]:
total_profit

800

## Evaluating conditions using Python

Apart from arithmetic operations, Python also provides several operations for comparing numbers & variables.

| Operator    | Description                                                     |
|-------------|-----------------------------------------------------------------|
| `==`        | Check if operands are equal                                     |
| `!=`        | Check if operands are not equal                                 |
| `>`         | Check if left operand is greater than right operand             |
| `<`         | Check if left operand is less than right operand                |
| `>=`        | Check if left operand is greater than or equal to right operand |
| `<=`        | Check if left operand is less than or equal to right operand    |

The result of a comparison operation is either `True` or `False` (note the uppercase `T` and `F`). These are special keywords in Python. Let's try out some experiments with comparison operators.

In [28]:
my_favorite_number = 1
my_least_favorite_number = 5
a_neutral_number = 3

In [29]:
# Equality check - True
my_favorite_number == 1

True

In [30]:
# Equality check - False
my_favorite_number == my_least_favorite_number

False

In [31]:
# Not equal check - True
my_favorite_number != a_neutral_number

True

In [32]:
# Not equal check - False
a_neutral_number != 3

False

In [33]:
# Greater than check - True
my_least_favorite_number > a_neutral_number

True

In [34]:
# Less than check - True
my_favorite_number < 10

True

In [35]:
# Less than check - False
my_least_favorite_number < my_favorite_number

False

In [36]:
# Greater than or equal check - True
my_favorite_number >= 1

True

In [37]:
# Greater than or equal check - False
my_favorite_number >= 3

False

In [38]:
# Less than or equal check - True
3 + 6 <= 9

True

In [39]:
# Less than or equal check - False
my_favorite_number + a_neutral_number <= 3

False

Just like arithmetic operations, the result of a comparison operation can also be stored in a variable.

In [40]:
cost_of_ice_bag = 1.25
is_ice_bag_expensive = cost_of_ice_bag >= 10
print("Is the ice bag expensive?", is_ice_bag_expensive)

Is the ice bag expensive? False


## Combining conditions with logical operators

The logical operators `and`, `or` and `not` operate upon conditions and `True` & `False` values (also known as *booleans*). `and` and `or` operate on two conditions, whereas `not` operates on a single condition.

The `and` operator returns `True` when both the conditions evaluate to `True`. Otherwise, it returns `False`.

| `a`     | `b`    | `a and b` |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `False`   |
|  `False`| `True` | `False`   |
|  `False`| `False`| `False`   |

In [41]:
my_favorite_number

1

In [42]:
my_favorite_number > 0 and my_favorite_number <= 3

True

In [43]:
my_favorite_number < 0 and my_favorite_number <= 3

False

In [44]:
my_favorite_number > 0 and my_favorite_number >= 3

False

In [45]:
True and False

False

In [46]:
True and True

True

The `or` operator returns `True` if at least one of the conditions evaluates to `True`. It returns `False` only if both conditions are `False`.

| `a`     | `b`    | `a or b`  |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `True`    |
|  `False`| `True` | `True`    |
|  `False`| `False`| `False`   |


In [47]:
a_neutral_number = 3

In [48]:
a_neutral_number == 3 or my_favorite_number < 0

True

In [49]:
a_neutral_number != 3 or my_favorite_number < 0

False

In [50]:
my_favorite_number < 0 or True

True

In [51]:
False or False

False

The `not` operator returns `False` if a condition is `True` and `True` if the condition is `False`.

In [52]:
not a_neutral_number == 3

False

In [53]:
not my_favorite_number < 0

True

In [54]:
not False

True

In [55]:
not True

False

Logical operators can be combined to form complex conditions. Use round brackets or parentheses `(` and `)` to indicate the order in which logical operators should be applied.

In [56]:
(2 > 3 and 4 <= 5) or not (my_favorite_number < 0 and True)

True

In [57]:
not (True and 0 < 1) or (False and True)

False

If parentheses are not used, logical operators are applied from left to right.

In [59]:
not True and 0 < 1 or False and True

False